lib/mongodb.js

```diff
@@ -66,8 +66,12 @@ exports.initialize = function initializeDataSource(dataSource, callback) {
   dataSource.connector = new MongoDB(s, dataSource);
   dataSource.ObjectID = mongodb.ObjectID;
 
-  if (callback) {
-    dataSource.connector.connect(callback);
+  if (s.lazyConnect) {
+    return true;
```

Comment:
I temporarily `return true` here since I need investigate more is it safe to execute [`postInit`](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/datasource.js#L315-L332) without trying connection. 
If it doesn't use any results from `dataSource.connector.connect` then we can just `callback` here.
Otherwise I might need to pass in some data in `callback(err, data)`